routing/missioncontrol_store_test.go

```diff
@@ -84,37 +82,40 @@ func TestMissionControlStore(t *testing.T) {
 
 	failureSourceIdx := 1
 
-	result1 := paymentResult{
-		route:            &mcStoreTestRoute,
-		failure:          lnwire.NewFailIncorrectDetails(100, 1000),
-		failureSourceIdx: &failureSourceIdx,
-		id:               99,
-		timeReply:        testTime,
-		timeFwd:          testTime.Add(-time.Minute),
-	}
+	result1 := newPaymentResult(
+		99, mcStoreTestRoute, testTime, testTime,
+		newPaymentFailure(
+			&failureSourceIdx,
+			lnwire.NewFailIncorrectDetails(100, 1000),
+		),
+	)
 
-	result2 := result1
-	result2.timeReply = result1.timeReply.Add(time.Hour)
-	result2.timeFwd = result1.timeReply.Add(time.Hour)
-	result2.id = 2
+	result2 := newPaymentResult(
```

Comment:
Can we also add a test with a failure that has no details? Just to show that works too (didn't see one, but perhaps I didn't look hard enough).

Reply:
indeed - good idea! added